### PR TITLE
Feat/new page folder

### PR DESCRIPTION
**Pull request by gweiying**
This PR adds functionality to create folder, subfolder and unlinked pages, based on the related issue #356. The functionality to edit these page settings will be added in a later PR. This PR should be reviewed with [#122](https://github.com/isomerpages/isomercms-backend/pull/122) on the backend.

Preview of Page Setting modal when used for creating new page with duplicated title as current files:
Screenshot 2021-03-08 at 6 48 26 PM

## Notes
- While folder, subfolder and unlinked pages can all be created right now, the redirection to subfolder pages will not work as we currently do not support editing subfolder pages via EditPage on the backend. We should add this functionality in a separate PR. Issue #361
- This PR also does not load the website url from the config page, as we plan to add the website url to the display on the leftbar. Once that is completed, we will have access to this value throughout the site. Issue #372 
- ~Our collection page data retrieved from the `collection.yml` via `convertFolderOrderToArray` and `convertSubfolderArray` has a different format (`{title, type, path}`) from our unlinked page data retrieved via `Directory.list()` on the backend ({`name`, `path`, `sha`, `size`, `type`}). We should consider standardising the name of attributes with the same value i.e. `title`, and `name`.~ fixed with f86253f

### CODE REVIEW

File: src/components/PageSettingsModal.jsx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    const { setRedirectToPage } = useRedirectHook()
+
+    const idToSetterFuncMap = {
+      title: setTitle,
+      permalink: setPermalink,
+    }
+
+    const { data } = useQuery(
+      'page',
+      () => console.log('getting page data'),
```
Comment (alexanderleegs) [inline src/components/PageSettingsModal.jsx:58]: Nit: could remove log statement

**Review by alexanderleegs — CHANGES_REQUESTED** [review]
Left a comment on a log statement. There is also a slight display issue with the permalink field:
Screenshot 2021-03-10 at 12 32 14 PM

File: src/components/PageSettingsModal.jsx (hunk 1 above)
Comment (gweiying) [reply src/components/PageSettingsModal.jsx]: Fixed with 7ce1332

**Comment by gweiying** [comment]
> Left a comment on a log statement. There is also a slight display issue with the permalink field:
> Screenshot 2021-03-10 at 12 32 14 PM

Good catch, fixed with d69811a

**Review by alexanderleegs — APPROVED** [review]
lgtm!

**Pull request merged**